.github/workflows/clang_format.yml

```diff
@@ -0,0 +1,34 @@
+name: Clang Format
+
+on:
+  workflow_dispatch:
+  push:
+    branches: ["main"]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: ["main"]
+
+jobs:
+  clang-format:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Install Clang Format
+        run: |
+          # TODO: Figure out how always to install the latest version of clang-format
```

Comment:
There isn't clang-format major release that often, I think is fine if we have to manually upgrade the version from time to time when we require a feature from a more recent version.